x/gov/handler_sidechain.go

```diff
@@ -6,6 +6,16 @@ import (
 )
 
 func handleMsgSideChainSubmitProposal(ctx sdk.Context, keeper Keeper, msg MsgSideChainSubmitProposal) sdk.Result {
+	if sdk.IsUpgrade(sdk.BCFusionSecondHardFork) {
+		return sdk.ErrMsgNotSupported("").Result()
+	}
+	if sdk.IsUpgrade(sdk.BCFusionFirstHardFork) {
+		vp := keeper.vs.GetSideChainTotalVotingPower(ctx, msg.SideChainId)
```

Comment:
Before you do `GetSideChainTotalVotingPower`, you need to do `keeper.ScKeeper.PrepareCtxForSideChain(ctx, msg.SideChainId)`.

In this way, you dont need to pass the `msg.SideChainId` as a parameter.

Reply:
fixed

Reply:
`PrepareCtxForSideChain` done twice here, no need